src/clib/pioc_support.c

```diff
@@ -4078,18 +4092,26 @@ int pioc_change_def(int ncid, int is_enddef)
                 if (file->iotype == PIO_IOTYPE_NETCDF)
                 {
 #ifdef NETCDF_C_NC__ENDDEF_EXISTS
-                    /* Sets the pad at the end of the "header" section. */
-                    const size_t h_minfree = 10 * 1024; /* The recommended size by Charlie Zender (NCO developer) is 10 KB */
+                    if (file->reserve_extra_header_space)
+                    {
+                        /* Sets the pad at the end of the "header" section.
+                         * The recommended size by Charlie Zender (NCO developer) is 10 KB */
+                        const MPI_Offset h_minfree = (PIO_RESERVED_FILE_HEADER_SIZE >= 0)? PIO_RESERVED_FILE_HEADER_SIZE : (10 * 1024);
```

Comment:
Same as above - change the sanity check to an assert

Reply:
OK, I have made this change.